rustls/src/crypto/signer.rs

```diff
@@ -61,6 +62,12 @@ pub trait SigningKey: Debug + Send + Sync {
 
     /// What kind of key we have.
     fn algorithm(&self) -> SignatureAlgorithm;
+
+    /// Get the RFC 5280-compliant SubjectPublicKeyInfo (SPKI) of this [`SigningKey`] if it exists.
+    fn public_key(&self) -> Result<Option<Vec<u8>>, Error> {
```

Comment:
Is there a reason to imagine that someone might implement this to return `Ok(None)`? It feels like `Result<Vec<u8>, Error>` might be sufficient and make this a little more ergonomic for a caller.

Reply:
Can disregard this based on the idea of using `Ok(None)` for the unimplemented case.